apps/rush-lib/src/logic/pnpm/PnpmLinkManager.ts

```diff
@@ -222,11 +222,17 @@ export class PnpmLinkManager extends BaseLinkManager {
     //   C%3A%2Fwbt%2Fcommon%2Ftemp%2Fprojects%2Fapi-documenter.tgz
     //   C%3A%2Fdev%2Fimodeljs%2Fimodeljs%2Fcommon%2Ftemp%2Fprojects%2Fpresentation-integration-tests.tgz_jsdom@11.12.0
     //   C%3A%2Fdev%2Fimodeljs%2Fimodeljs%2Fcommon%2Ftemp%2Fprojects%2Fbuild-tools.tgz_2a665c89609864b4e75bc5365d7f8f56
-    const folderNameInLocalInstallationRoot: string =
-      uriEncode(Text.replaceAll(absolutePathToTgzFile, path.sep, '/')) + folderNameSuffix;
+    let folderNameInLocalInstallationRoot: string =
+      uriEncode(Path.convertToSlashes(absolutePathToTgzFile)) + folderNameSuffix;
+
+    // PNPM 6 changed formatting to replace all special chars with '+'
+    // e.g.: C++dev+imodeljs+imodeljs+common+temp+projects+presentation-integration-tests.tgz_jsdom@11.12.0
+    if (this._pnpmVersion.major >= 6) {
+      const specialCharRegex: RegExp = /%[a-fA-FA-F0-9]{2}/g;
+      folderNameInLocalInstallationRoot = folderNameInLocalInstallationRoot.replace(specialCharRegex, '+');
+    }
 
     // e.g.: C:\wbt\common\temp\node_modules\.local\C%3A%2Fwbt%2Fcommon%2Ftemp%2Fprojects%2Fapi-documenter.tgz\node_modules
```

Comment:
@D4N14L When checking out your branch, I encountered an error since your GitHub fork is still called `web-build-tools`.  Might be time to update that.  :-)